### DISCUSSION TITLE
Need help updating an old package having export = temp in its module

### DISCUSSION THREAD

**Discussion by fungiboletus** [DefinitelyTyped/DefinitelyTyped#56205] [_Misc]
Hei,

I want to update a definition that hasn't been updated since many years, but I can't find the right magical syntax to let it compile and not break backward compatibility. It has a namespace but also a function with a constructor by default.

It's used like this in the tests:

```javascript
import Busboy = require('busboy')
var busboy = new Busboy({...stuff});
```

The definitions look like this:
```typescript

declare namespace busboy {
    ...
    interface BusboyConstructor {
        new (options: BusboyConfig): Busboy;
    }
}

declare module 'busboy' {
    const temp: busboy.BusboyConstructor;
    export = temp;
}
```

This is not accepted nor valid anymore it seems, running the tests produces the following error with unhelpful help messages:

```
ERROR: 9:1   strict-export-declare-modifiers  'declare' keyword is redundant here. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 9:19  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 61:5  expect                           TypeScript@4.5 compile error: 
Exports and export assignments are not permitted in module augmentations.
```

I tried the following and it compiles but the namespace is not exported anymore:

```typescript

declare namespace busboy {
    ...
    interface BusboyConstructor {
        new (options: BusboyConfig): Busboy;
    }
}

declare const temp: busboy.BusboyConstructor;
export = temp;
```

I'm a bit lost because I can't find a similar commonjs module that has the right syntax to export both a default interface with a constructor, and a namespace. Is it possible ? If it's not possible, what's the closest I can suggest ?

Thanks.

**Comment by hozblok** [comment]
Hi @fungiboletus ,

There is a trick. You can use the same name for the namespace and for the const as [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f7ec78508c6797e42f87a4390735bc2c650a1bfd/types/es-abstract/index.d.ts):

```typescript
declare namespace busboy {
    ...
    interface BusboyConstructor {
        new (options: BusboyConfig): Busboy;
    }
}

declare const busboy: busboy.BusboyConstructor;
export = busboy;
```
